.buildkite/hooks/post-checkout

```diff
@@ -32,6 +34,49 @@ checkout_merge() {
     }
 }
 
+retry() {
+    local retries=$1
+    shift
+
+    local count=0
+    until "$@"; do
+        exit=$?
+        wait=$((2 ** count))
+        count=$((count + 1))
+        if [ $count -lt "$retries" ]; then
+            >&2 echo "Retry $count/$retries exited $exit, retrying in $wait seconds..."
+            sleep $wait
+        else
+            >&2 echo "Retry $count/$retries exited $exit, no more retries left."
+            return $exit
+        fi
+    done
+    return 0
+}
+
+GITHUB_REPO_TOKEN=$(retry 5 vault kv get -field token ${GITHUB_TOKEN_VAULT_PATH})
+
+# Usage:
+#check_if_file_exist_in_repo "infra" "main" "cd/release/release-manager/project-configs" "build.gradle"
+#Returns IS_FILE_EXISTS_IN_REPO=true if file exists and IS_FILE_EXISTS_IN_REPO=false if not exists
+check_if_file_exist_in_repo() {
+    local repoName=${1}
+    local branch=${2}
+    local fileName="build.gradle"
+
+    if [[ $branch == "main" ]]; then
+        local branchName="master"
+    fi
+
+    local path_to_file="cd/release/release-manager/project-configs/${branchName}/${fileName}"
+    local response=$(curl -s -u $GITHUB_REPO_TOKEN:x-oauth-basic https://api.github.com/repos/elastic/${repoName}/contents/${path_to_file}| grep -c "\"path\"\: \"${path_to_file}\"")
+    if [[ ${response} -ge 1 ]]; then
+        export IS_FILE_EXISTS_IN_REPO=true
+    else
+        export IS_FILE_EXISTS_IN_REPO=false
```

Comment:
This variable is the same as this one ? 
https://github.com/elastic/fleet-server/blob/331112b16439ccfcef7883e49baf5ac325515601/.ci/packaging.groovy#L122

Why not calling it as in Jenkins `IS_BRANCH_AVAILABLE`?

Reply:
It was renamed because, in fact, it's checking a file in the repository, not a branch.